hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClusterConnectorServiceImpl.java

```diff
@@ -127,45 +127,44 @@ public void connectToCluster() {
 
     @Override
     public boolean isClusterAvailable() {
-        return getOwnerConnectionAddress() != null;
+        return clusterListeningConnection != null;
     }
 
-    public Address getOwnerConnectionAddress() {
-        return ownerConnectionAddress;
+    @Override
+    public ClientConnection getClusterConnection() {
+        return clusterListeningConnection;
     }
 
-    public void setOwnerConnectionAddress(Address ownerConnectionAddress) {
-        if (this.ownerConnectionAddress != null) {
-            this.previousOwnerConnectionAddress = this.ownerConnectionAddress;
-        }
-        this.ownerConnectionAddress = ownerConnectionAddress;
+    private void setClusterConnection(ClientConnection connection) {
+        clusterListeningConnection = connection;
     }
 
-    private Connection connectAsOwner(Address address) {
+    private Connection connect(Address address) {
         Connection connection = null;
         try {
-            logger.info("Trying to connect to " + address + " as owner member");
-            connection = connectionManager.getOrConnect(address, true);
-            setOwnerConnectionAddress(connection.getEndPoint());
+            logger.info("Trying to connect to " + address + " as main member");
```

Comment:
We do not want to introduce a new concept of `what is main member` in place of `owner member`. Hence, I suggest that if active connections is 0 and the first connection (0->1) will add the cluster listeners.